src/Razor/test/Microsoft.NET.Sdk.Razor.Test/GenerateAspNetCoreStaticAssetsManifestTest.cs

```diff
@@ -0,0 +1,280 @@
+
+            var task = new GenerateAspNetCoreStaticAssetsManifest
+            {
+                BuildEngine = buildEngine.Object,
+                ContentRootDefinitions = new TaskItem[]{
+                    CreateItem(Path.Combine("wwwroot","sample.js"), new Dictionary<string,string>{
+                        ["BasePath"] = "MyLibrary",
+                        ["ContentRoot"] = Path.Combine(".", "MyLibrary")
+                    }),
+                    CreateItem(Path.Combine("wwwroot", "otherLib.js"), new Dictionary<string,string>{
+                        ["BasePath"] = "MyOtherLibrary",
+                        ["ContentRoot"] = Path.Combine(".", "MyLibrary")
+                    })
+                }
+            };
+
+            // Act
+            var result = task.Execute();
+
+            // Assert
+            Assert.False(result);
+            var message = Assert.Single(errorMessages);
+            Assert.Equal(
+                $"Duplicate content root paths '{Path.Combine(".", "MyLibrary")}' for base paths 'MyOtherLibrary' and 'MyLibrary' " +
+                $"('{Path.Combine("wwwroot", "otherLib.js")}', '{Path.Combine("wwwroot", "sample.js")}')",
+                message);
+        }
+
+        [Fact]
+        public void Generates_EmptyManifest_WhenNoItems_Passed()
+        {
+            // Arrange
+            var file = Path.GetTempFileName();
+            var expectedDocument = @"<AspNetCoreStaticAssets Version=""1.0"" />";
+
+            try
+            {
+                var buildEngine = new Mock<IBuildEngine>();
+
+                var task = new GenerateAspNetCoreStaticAssetsManifest
+                {
+                    BuildEngine = buildEngine.Object,
+                    ContentRootDefinitions = new TaskItem[] { },
+                    TargetManifestPath = file
+                };
+
+                // Act
+                var result = task.Execute();
+
+                // Assert
+                Assert.True(result);
+                var document = File.ReadAllText(file);
+                Assert.Equal(expectedDocument, document);
+            }
+            finally
+            {
+                if (File.Exists(file))
+                {
+                    File.Delete(file);
+                }
+            }
+        }
+
+        [Fact]
+        public void Generates_Manifest_WhenContentRootsAvailable()
+        {
+            // Arrange
+            var file = Path.GetTempFileName();
+            var expectedDocument = $@"<AspNetCoreStaticAssets Version=""1.0"">
+  <ContentRoot BasePath=""MyLibrary"" Path=""{Path.Combine(".", "nuget", "MyLibrary", "razorContent")}"" />
+</AspNetCoreStaticAssets>";
+
+            try
+            {
+                var buildEngine = new Mock<IBuildEngine>();
+
+                var task = new GenerateAspNetCoreStaticAssetsManifest
+                {
+                    BuildEngine = buildEngine.Object,
+                    ContentRootDefinitions = new TaskItem[] {
```

Comment:
format this. It looks like javascript.

Reply:
You make it sound as if it was a bad thing :p. Will fix